dialect.go

```diff
@@ -49,6 +51,9 @@ type Dialect interface {
 	// string to truncate tables
 	TruncateClause() string
 
+	// string to sleep for d duration
+	SleepClause(d time.Duration) string
```

Comment:
This is a breaking change and will force a `v3` of `gorp`, since some `gorp` users may have implemented custom `Dialect` types.  Can we instead add a new interface type, say `SleepDialect`, such that:

```go
type SleepDialect interface {
    SleepClause(time.Duration) string
}
```

And then remove the `SleepClause` methods on dialects that don't have a sleep clause?

Reply:
In addition, since this method is *only* used by tests, can we move the `SleepDialect` interface to the test package?  If it's in `gorp`'s main package, then users expect that implementing `SleepDialect` gives them something from `gorp`, rather than `gorp`'s tests.